packages/smooth_app/lib/helpers/analytics_helper.dart

```diff
@@ -0,0 +1,263 @@
+import 'package:smooth_app/database/dao_int.dart';
+import 'package:smooth_app/database/local_database.dart';
+
+// TODO(m123): Check for user consent
+// TODO(m123): handle debug mode
+
+/// Helper for logging usage of core features and exceptions
+/// Logging:
+/// - Errors and Problems (sentry)
+/// - App start
+/// - Product scan
+/// - Product page open
+/// - Knowledge panel open
+/// - personalized ranking (without sharing the preferences)
+/// - search
+/// - external links
+
+const String _initAction = 'started app';
+const String _scanAction = 'scanned product';
+const String _productPageAction = 'opened product page';
+const String _knowledgePanelAction = 'opened knowledge panel page';
+const String _personalizedRankingAction = 'personalized ranking';
+const String _searchAction = 'search';
+const String _linkAction = 'opened link';
+
+/// The event category. Must not be empty. (eg. Videos, Music, Games...)
+const String eventCategory = 'e_c';
+
+/// Must not be empty. (eg. Play, Pause, Duration, Add
+/// Playlist, Downloaded, Clicked...)
+const String eventAction = 'e_a';
+
+/// The event name. (eg. a Movie name, or Song name, or File name...)
+const String eventName = 'e_n';
+
+/// Must be a float or integer value (numeric), not a string.
+const String eventValue = 'e_v';
+
+String latestSearch = '';
+
+Future<void> initSentry({Function()? appRunner}) async {
+  final PackageInfo packageInfo = await PackageInfo.fromPlatform();
+
+  await SentryFlutter.init(
+    (SentryOptions options) {
+      options.dsn =
+          'https://22ec5d0489534b91ba455462d3736680@o241488.ingest.sentry.io/5376745';
+      options.sentryClientName = 'sentry.dart.smoothie/${packageInfo.version}';
+    },
+    appRunner: appRunner,
+  );
+}
+
+Future<void> initMatomo(
+  final BuildContext context,
+  final LocalDatabase _localDatabase,
+) async {
+  MatomoForever.init(
+    'https://analytics.openfoodfacts.org/matomo.php',
+    2,
+    id: _getId(),
+    // If we track or not, should be decidable later
+    rec: true,
+    method: MatomoForeverMethod.post,
+    sendImage: false,
+    // 32 character authorization key used to authenticate the API request
+    // only needed for request which are more then 24h old
+    // tokenAuth: 'xxx',
+  );
+}
+
+Future<bool> trackStart(
+    LocalDatabase _localDatabase, BuildContext context) async {
+  final TrackingHelper trackingHelper = TrackingHelper(_localDatabase);
+  final Map<String, String> data = <String, String>{};
+
+  data.addIfVAndNew('_idvc', trackingHelper.getAppVisits().toString());
+  data.addIfVAndNew(
+    '_viewts',
+    trackingHelper.getPreviousVisitUnix().toString(),
+  );
+  data.addIfVAndNew(
+    '_idts',
+    trackingHelper.getFirstVisitUnix().toString(),
+  );
+  data.addIfVAndNew('res', MediaQuery.of(context).size.toString());
+  data.addIfVAndNew('lang', Localizations.localeOf(context).languageCode);
+  data.addIfVAndNew('country', Localizations.localeOf(context).countryCode);
```

Comment:
Please do something about those "magic strings": I have no way to understand what '_idts' stands for and where it comes from.

Reply:
Added some more comments